packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -11,6 +11,120 @@ import 'package:flutter_test/flutter_test.dart';
 const Offset _kRowOffset = Offset(0.0, -32.0);
 
 void main() {
+  testWidgets('time picker scrolls to 00:00:00 after reset', (WidgetTester tester) async {
+    const Duration animationTime = Duration(milliseconds: 200);
+    final CupertinoTimerPickerController controller = CupertinoTimerPickerController(
+        resetAnimationDuration: animationTime,
+    );
+    Duration duration;
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Padding(
+          padding: const EdgeInsets.all(40.0),
+          child: Column(
+            children: <Widget>[
+              Container(
+                height: 300,
+                child: CupertinoTimerPicker(
+                  controller: controller,
+                  onTimerDurationChanged: (Duration newDuration) { duration = newDuration; },
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    await tester.drag(find.text('0').first, _kRowOffset * 2);
+    await tester.drag(find.text('0').at(1), _kRowOffset * 3);
+    await tester.drag(find.text('0').last, _kRowOffset * 4);
+
+    expect(duration, const Duration(hours: 2, minutes: 3, seconds: 4));
+
+    controller.desiredTimerDuration = const Duration();
+    await tester.pump();
+    await tester.pump(animationTime);
```

Comment:
Are 2 pump()s necessary?  Maybe just a pumpAndSettle()?

Reply:
Apparently we are supposed to do this instead of a pumpAndSettle because it's more precise.